beacon_node/eth1/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use ethabi::Error as AbiError;
+use ssz;
+use web3;
+
+#[derive(Debug)]
+pub enum Error {
+    /// Errors that occur on interaction with an Eth1 node.
+    Web3Error(web3::error::Error),
+    /// Web3 request timeout.
+    Timeout,
+    /// Error on interacting with a contract on Eth1.
+    ContractError(web3::contract::Error),
+    /// Decoding error.
+    DecodingError,
+    /// Invalid parameters
+    InvalidParam,
+    /// Invalid data
+    InvalidData,
```

Comment:
Perhaps this could also be more detailed?

Reply:
Removed these types. Most of the error types should have been variants of `web3::Error` or `web3::Contract::Error`